src/materials/denoiser/createAlbedoMaterial.js

```diff
@@ -0,0 +1,128 @@
+import { Color, ShaderMaterial } from 'three';
+
+export function createAlbedoShaderMaterial( originalMaterial ) {
```

Comment:
Is there a reason we're not just using `MeshBasicMaterial` here? Once you start including things like metalness and  clearcoat and other specular terms you're no longer just rendering albedo.

Reply:
We can but technically it would be incorrect (at least according to OIDN)
[Here in the docs](https://github.com/RenderKit/oidn?tab=readme-ov-file#albedos) They say:
> The albedo image is the feature image that usually provides the biggest quality improvement. It should contain the approximate color of the surfaces independent of illumination and viewing angle.

> For simple matte surfaces this means using the diffuse color/texture as the albedo. For other, more complex surfaces it is not always obvious what is the best way to compute the albedo, but the denoising filter is flexible to a certain extent and works well with differently computed albedos. Thus it is not necessary to compute the strict, exact albedo values but must be always between 0 and 1.

> For metallic surfaces the albedo should be either the reflectivity at normal incidence (e.g. from the artist friendly metallic Fresnel model) or the average reflectivity; or if these are constant (not textured) or unknown, the albedo can be simply 1 as well.

> The albedo for dielectric surfaces (e.g. glass) should be either 1 or, if the surface is perfect specular (i.e. has a delta BSDF), the Fresnel blend of the reflected and transmitted albedos. The latter usually works better but only if it does not introduce too much noise or the albedo is prefiltered. If noise is an issue, we recommend to split the path into a reflected and a transmitted path at the first hit, and perhaps fall back to an albedo of 1 for subsequent dielectric hits. The reflected albedo in itself can be used for mirror-like surfaces as well.

> The albedo for layered surfaces can be computed as the weighted sum of the albedos of the individual layers. Non-absorbing clear coat layers can be simply ignored (or the albedo of the perfect specular reflection can be used as well) but absorption should be taken into account.

Then regarding transparency and reflective items you can either be safe and just set it to 1 or use a Fresnel blend of the reflected and transmitted albedos.

`MeshBasic` will be correct for matte objects and might be ok for others, but this shader was an attempt to accommodate the other cases (metallic, roughness,  clearcoat, etc)

Besides the sky, I don't see much issues with this actually. It probably can be improved, a pathtraced albedo would be even better but would slow things down.

Totally up to you though.

Reply:
The shader is being problematic and I get good results with meshBasic material. I'm removing the shader for now and will think about a strategy for transmission later